src/Ormolu/Printer/Combinators.hs

```diff
@@ -108,6 +109,27 @@ located' ::
   R ()
 located' = flip located
 
+-- | A version of 'located' that works on 'Pat'.
+--
+-- Starting from GHC 8.8, 'LPat' == 'Pat'. Located 'Pat's are always
+-- constructed with the 'XPat' constructor, containing a @Located Pat@.
+--
+-- Most of the time, we can just use 'p_pat' directly, because it handles
+-- located 'Pat's. However, sometimes we want to use the location to render
+-- something other than the given 'Pat'.
+--
+-- If given 'Pat' does not contain a location, we error out.
```

Comment:
I'm rather disappointed with GHC here.

Of course, an alternative option is swallowing the bug and doing `p -> ($ p)` if we get something other than an `XPat`. I'm not sure what is the best course of action here.